### ISSUE TITLE
DIoU loss implementation

### ISSUE THREAD

**Issue by simon-rob** [wlguan/Stronger-yolo-pytorch#10]
https://github.com/Zzh-tju/DIoU-darknet
https://github.com/Zzh-tju/DIoU-SSD-pytorch
https://github.com/Zzh-tju/DIoU-pytorch-detectron
This already implemented in the above repositories.

loss | AP
iou loss | 46.57
giou loss | 47.73
diou loss | 48.10
ciou loss | 49.21

**Comment by wlguan** [comment]
@simon-rob  Thanks, maybe add label next time.

**Comment by wlguan** [comment]
@simon-rob  Hi, I have added the diou-loss, but the ciouloss will result in Nan loss. According to my result, DIOU got 79.0 for mAP50, maybe we should tune the hyperparameter?  You may try it yourself.
